--- include/dune/grid/io/file/dgfparser/blocks/periodicfacetrans.hh.orig	2026-02-16 11:06:08.587957357 +0000
+++ include/dune/grid/io/file/dgfparser/blocks/periodicfacetrans.hh	2026-02-16 11:07:46.804485236 +0000
@@ -23,10 +23,54 @@
     struct PeriodicFaceTransformationBlock
       : public BasicBlock
     {
-      template< class T >
-      class Matrix;
-
-      struct AffineTransformation;
+        template< class T >
+        class Matrix
+        {
+          int rows_;
+          int cols_;
+          std::vector< T > fields_;
+
+        public:
+          Matrix ( int rows, int cols )
+            : rows_( rows ),
+              cols_( cols ),
+              fields_( rows * cols )
+          {}
+
+          const T &operator() ( int i, int j ) const
+          {
+            return fields_[ i * cols_ + j ];
+          }
+
+          T &operator() ( int i, int j )
+          {
+            return fields_[ i * cols_ + j ];
+          }
+
+          int rows () const
+          {
+            return rows_;
+          }
+
+          int cols () const
+          {
+            return cols_;
+          }
+        };
+
+        // PeriodicFaceTransformationBlock::AffineTransformation
+        // -----------------------------------------------------
+
+        struct AffineTransformation
+        {
+          Matrix< double > matrix;
+          std::vector< double > shift;
+
+          explicit AffineTransformation ( int dimworld )
+            : matrix( dimworld, dimworld ),
+              shift( dimworld )
+          {}
+        };
 
     private:
       std::vector< AffineTransformation > transformations_;
@@ -53,61 +97,6 @@
       void match ( char what );
     };
 
-
-    // PeriodicFaceTransformationBlock::Matrix
-    // ---------------------------------------
-
-    template< class T >
-    class PeriodicFaceTransformationBlock::Matrix
-    {
-      int rows_;
-      int cols_;
-      std::vector< T > fields_;
-
-    public:
-      Matrix ( int rows, int cols )
-        : rows_( rows ),
-          cols_( cols ),
-          fields_( rows * cols )
-      {}
-
-      const T &operator() ( int i, int j ) const
-      {
-        return fields_[ i * cols_ + j ];
-      }
-
-      T &operator() ( int i, int j )
-      {
-        return fields_[ i * cols_ + j ];
-      }
-
-      int rows () const
-      {
-        return rows_;
-      }
-
-      int cols () const
-      {
-        return cols_;
-      }
-    };
-
-
-    // PeriodicFaceTransformationBlock::AffineTransformation
-    // -----------------------------------------------------
-
-    struct PeriodicFaceTransformationBlock::AffineTransformation
-    {
-      Matrix< double > matrix;
-      std::vector< double > shift;
-
-      explicit AffineTransformation ( int dimworld )
-        : matrix( dimworld, dimworld ),
-          shift( dimworld )
-      {}
-    };
-
-
     inline std::ostream &
     operator<< ( std::ostream &out, const PeriodicFaceTransformationBlock::AffineTransformation &trafo )
     {
